File: dayflow-hrms/backend/src/main/java/com/dayflow/hrms/service/AuthService.java
package com.dayflow.hrms.service;

import com.dayflow.hrms.dto.auth.AuthResponse;
import com.dayflow.hrms.dto.auth.LoginRequest;
import com.dayflow.hrms.dto.auth.SignUpRequest;
import com.dayflow.hrms.security.UserPrincipal;

public interface AuthService {
    AuthResponse signUp(SignUpRequest request, String ipAddress);
    AuthResponse signIn(LoginRequest request, String ipAddress);
    boolean verifyEmail(String token);
    AuthResponse getCurrentUser(UserPrincipal currentUser);
}

File: dayflow-hrms/frontend/src/services/authService.js
import api from './api';

export const authService = {
  login: (credentials) => api.post('/auth/login', credentials),
  signup: (userData) => api.post('/auth/signup', userData),
  verifyEmail: (token) => api.post('/auth/verify-email', { token }),
  getCurrentUser: () => api.get('/auth/me'),
  logout: () => api.post('/auth/logout')
};
